pkg/resources/ali/ali.go

```diff
@@ -689,3 +693,138 @@ func DeleteVolume(cloud fi.Cloud, r *resources.Resource) error {
 	}
 	return nil
 }
+
+func ListNatGateway(d *clusterDiscoveryALI, vpcID string, resourceTrackers *[]*resources.Resource) error {
+
+	// Delete NateGateway with specified name. All of the EIPs will be deleted.
```

Comment:
Any tagging you can do to be more confident is great, but at least on AWS & GCP not all objects can be tagged.  (But I think on AWS we're down to just SSH keys not being taggable!)